content/en/docs/concepts/scheduling-eviction/assign-pod-node.md

```diff
@@ -234,10 +234,38 @@ Pods, the default Kubernetes scheduler places those Pods and honors any
 ### Inter-pod affinity and anti-affinity
 
 Inter-pod affinity and anti-affinity allow you to constrain which nodes your
-Pods can be scheduled on based on the labels of **Pods** already running on that
+Pods can be scheduled on based on the labels of Pods already running on that
 node, instead of the node labels.
 
-Inter-pod affinity and anti-affinity rules take the form "this
+#### Existing Pods' Affinity and Anti-Affinity Considerations
+
+When scheduling a new Pod, the Kubernetes scheduler evaluates both the incoming Pod's affinity/anti-affinity rules and those of the existing Pods in the cluster. Here's how these rules are processed:
+
+1. Hard Constraints (Node Filtering):
+   - Incoming Pod's `podAffinity.requiredDuringSchedulingIgnoredDuringExecution`:
+     - The scheduler ensures that the new Pod is scheduled onto a node that satisfies the required affinity rules based on the existing Pods.
+   - Incoming Pod's `podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution`:
+     - The scheduler filters out nodes that do not satisfy the required anti-affinity rules, ensuring that the new Pod does not get scheduled onto nodes where it would violate these rules.
+   - Existing Pods' `podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution`:
+     - The scheduler treats these as hard constraints, filtering out any nodes that do not comply with the anti-affinity requirements of existing Pods. This ensures that the new Pod does not end up on a node that would violate the anti-affinity rules of existing Pods.
+
+2. Soft Constraints (Scoring):
+   - Incoming Pod's `podAffinity.preferredDuringSchedulingIgnoredDuringExecution`:
+     - The scheduler scores nodes based on how well they satisfy the preferred affinity rules, aiming to place the new Pod in locations that optimize for these preferences.
+   - Incoming Pod's `podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution`:
+     - Similar to preferred affinity, the scheduler scores nodes to avoid anti-affinity preferences, aiming to minimize conflicts.
+   - Existing Pods' `podAffinity.requiredDuringSchedulingIgnoredDuringExecution`:
+     - The scheduler treats them as soft constraints, and scores nodes based on how well they satisfy the affinity requirements of existing Pods, promoting optimal placement without enforcing strict rules.
+
+3. Ignored Fields:
+   - Existing Pods' `podAffinity.preferredDuringSchedulingIgnoredDuringExecution`:
+     - These preferred affinity rules are not considered during the scheduling decision for new Pods.
+   - Existing Pods' `podAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution`:
+     - Similarly, preferred anti-affinity rules of existing Pods are ignored during scheduling.
```

Comment:
I'd suggest we remove line 260-264.
The affinity settings of "Existing Pods" are treated as data source when assessing the affinity settings of "Incoming Pods". They are not changed (i.e. ignored) during scheduling.

The point is ... we don't have any mechanisms to re-assess Pod affinity during execution. The current implementation was based on an over-design. In other words,
there is nothing "required" or "ignored" during execution.

Reply:
For more contexts (how the community wants to improve this affinity handling with "required-during-execution"), you can check https://github.com/kubernetes/enhancements/pull/4329/files

Reply:
> we don't have any mechanisms to re-assess Pod affinity during execution. The current implementation was based on an over-design. In other words,

We have one thing I know of: https://github.com/kubernetes-sigs/descheduler
(but as an interim approach it relies on the existing allowed values for this field).

Reply:
> https://github.com/kubernetes/website/pull/48083#discussion_r1804138708

That's for node affinity though. There aren't any open efforts for pod affinity.